newIDE/app/src/ObjectsRendering/PixiResourcesLoader.js

```diff
@@ -308,9 +310,21 @@ export default class PixiResourcesLoader {
     project: gdProject,
     resourceName: string
   ): Promise<THREE.THREE_ADDONS.GLTF> {
-    const loaded3DModel = loaded3DModels[resourceName];
-    if (loaded3DModel) return Promise.resolve(loaded3DModel);
+    const loading3DModelPromise = loading3DModelPromises[resourceName];
+    if (loading3DModelPromise) return loading3DModelPromise;
 
+    const loadingPromise = PixiResourcesLoader._load3DModel(
+      project,
+      resourceName
+    );
+    loading3DModelPromises[resourceName] = loadingPromise;
+    return loadingPromise;
+  }
+
+  static _load3DModel(
```

Comment:
Consider if this can be moved out of the class (and if it's worth it) if that's a pure function